From c5a50b67fb442f8eeb5274bd106de216db8884e2 Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Tue, 2 Aug 2022 15:51:03 +0200
Subject: [PATCH 0227/1695] dts: stm32: use non-secure I2C compatible on MP15
 board

Change i2c4 nodes of ST boards based on STM32MP15 to use compatible
st,stm32mp15-i2c-non-secure instead of st,stm32mp15-i2c. This is needed
by the coming firewall framework probe exception feature useless which
firewall framework prevents probing of non-secure reqources.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git cc14e11a217fdbaf73df33cc97a9670f8b31099e dts: stm32: use non-secure I2C compatible on MP15 board]
Change-Id: Ifc1c3e335b242ea77a6f12921668815c0376b99a
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
---
 core/arch/arm/dts/stm32mp157c-ed1.dts  | 1 +
 core/arch/arm/dts/stm32mp15xx-dkx.dtsi | 1 +
 2 files changed, 2 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp157c-ed1.dts b/core/arch/arm/dts/stm32mp157c-ed1.dts
index ba762e202..74797471e 100644
--- a/core/arch/arm/dts/stm32mp157c-ed1.dts
+++ b/core/arch/arm/dts/stm32mp157c-ed1.dts
@@ -175,6 +175,7 @@
 };
 
 &i2c4 {
+	compatible = "st,stm32mp15-i2c-non-secure";
 	pinctrl-names = "default", "sleep";
 	pinctrl-0 = <&i2c4_pins_a>;
 	pinctrl-1 = <&i2c4_sleep_pins_a>;
diff --git a/core/arch/arm/dts/stm32mp15xx-dkx.dtsi b/core/arch/arm/dts/stm32mp15xx-dkx.dtsi
index 3ebf6102d..188194135 100644
--- a/core/arch/arm/dts/stm32mp15xx-dkx.dtsi
+++ b/core/arch/arm/dts/stm32mp15xx-dkx.dtsi
@@ -238,6 +238,7 @@
 };
 
 &i2c4 {
+	compatible = "st,stm32mp15-i2c-non-secure";
 	pinctrl-names = "default", "sleep";
 	pinctrl-0 = <&i2c4_pins_a>;
 	pinctrl-1 = <&i2c4_sleep_pins_a>;
-- 
2.47.2

